code/modules/shuttle/shuttle_projector.dm

```diff
@@ -0,0 +1,169 @@
+/// Projects a shuttle with visual juice while it docks/launches with vis_contents
+/obj/effect/abstract/shuttle_projector
+	layer = LOWER_SHUTTLE_MOVEMENT_LAYER
+	plane = LOWER_SHUTTLE_MOVEMENT_PLANE
+	appearance_flags = KEEP_TOGETHER
+	mouse_opacity = MOUSE_OPACITY_TRANSPARENT
+	icon = 'icons/effects/alphacolors.dmi'
+	icon_state = "nothing"
+
+	/// The mobile port we're projecting
+	var/obj/docking_port/mobile/shuttle_port
+	/// The bottom left turf of the bounding box where the shuttle will dock in the stationary port
+	var/turf/bottom_left
+
+/obj/effect/abstract/shuttle_projector/Initialize(mapload, obj/docking_port/mobile/shuttle_port, obj/docking_port/stationary/stationary_port, inbound, total_animate_time = null)
+	. = ..()
+
+	if (!istype(shuttle_port))
+		stack_trace("Invalid shuttle_port for shuttle_projector!")
+		return INITIALIZE_HINT_QDEL
+	if (!istype(stationary_port))
+		stack_trace("Invalid stationary_port for shuttle_projector!")
+		return INITIALIZE_HINT_QDEL
+	src.shuttle_port = shuttle_port
+
+	// Get the mobile ports turfs to project
+	var/list/projected_turfs = shuttle_port.return_ordered_turfs(shuttle_port.x, shuttle_port.y, shuttle_port.z, shuttle_port.dir)
```

Comment:
If order of turfs unnecesary `return_turfs()` can be used here.

https://github.com/tgstation/tgstation/blob/8bf91013bf7fc12945da9ec361f230bb5225ff3f/code/modules/shuttle/shuttle.dm#L113-L146

Reply:
It better this way, so that in case someone builds a random grille, there's a higher chance of it being picked up automatically